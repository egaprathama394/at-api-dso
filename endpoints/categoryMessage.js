/* eslint no-unused-vars: off */
/* global env */
const supertest = require('supertest');
const env = require('dotenv').config();

process.env.NODE_TLS_REJECT_UNAUTHORIZED = 0;

const api = supertest(process.env.API_BASE_URL_MOBILE);

export default token => api.get('/api/Messages/category')
  .set('ClientTag', process.env.CLIENT_TAG)
  .set('ApplicationName', process.env.APPLICATION_NAME)
  .set('Authorization', `Bearer ${token}`)
  .set('server-code', process.env.SERVER_CODE_MOBILE)
  .set('Content-Type', 'application/json');
