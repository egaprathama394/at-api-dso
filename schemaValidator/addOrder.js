export default {
  type: 'object',
  required: [
    'orderNo',
    'acknowledge',
    'message',
    'version',
    'build',
  ],
  properties: {
    orderNo: {
      type: 'string',
    },
    acknowledge: {
      type: 'integer',
    },
    message: {
      type: 'string',
    },
    version: {
      type: 'string',
    },
    build: {
      type: 'string',
    },
  },
};

export const negativeReqSchema = {
  type: 'object',
  required: [
    'message',
  ],
  properties: {
    message: {
      type: 'string',
    },
  },
};

export const emptyReqInput = {
  type: 'object',
  required: [
    'message',
    'modelState',
  ],
  properties: {
    message: {
      type: 'string',
    },
    modelState: {
      type: 'object',
      required: [
        'request.BranchCode',
        'request.BookingDateTime',
        'request.VehicleType',
      ],
      properties: {
        'request.BranchCode': {
          type: 'array',
          items: {
            type: 'string',
          },
        },
        'request.BookingDateTime': {
          type: 'array',
          items: {
            type: 'string',
          },
        },
        'request.VehicleType': {
          type: 'array',
          items: {
            type: 'string',
          },
        },
      },
    },
  },
};
