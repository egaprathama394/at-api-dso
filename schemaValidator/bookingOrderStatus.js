export default {
  type: 'object',
  required: [
    'acknowledge',
    'message',
  ],
  properties: {
    acknowledge: {
      type: 'string',
    },
    message: {
      type: 'string',
    },
  },
};
