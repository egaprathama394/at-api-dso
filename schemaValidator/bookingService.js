export default {
  type: 'object',
  required: [
    'orderNo',
    'acknowledge',
    'message',
    'version',
    'build',
  ],
  properties: {
    orderNo: {
      type: 'string',
    },
    acknowledge: {
      type: 'integer',
    },
    message: {
      type: 'string',
    },
    version: {
      type: 'string',
    },
    build: {
      type: 'string',
    },
  },
};

export const displayBookingService = {
  type: 'object',
  required: [
    'bookingOrderList',
    'acknowledge',
    'version',
    'build',
  ],
  properties: {
    bookingOrderList: {
      type: 'array',
      items: {
        type: 'object',
        required: [
          'branchCode',
          'branchPhone1',
          'branchPhone2',
          'branchDescription',
          'branchAddress',
          'branchPICName',
          'orderNo',
          'orderStatus',
          'orderType',
          'totalAmount',
          'serviceOrderJobs',
          'orderStartTime',
          'orderEndTime',
          'policeRegNo',
          'vin',
          'vehicleType',
          'year',
          'name',
          'phone1',
          'phone2',
          'address',
          'serviceAddress',
          'notes',
          'dscProcessStatus',
          'actualFinishTime',
          'isEligibleForRating',
          'createdTime',
        ],
        properties: {
          branchCode: {
            type: 'string',
          },
          branchPhone1: {
            type: 'string',
          },
          branchPhone2: {
            type: 'string',
          },
          branchDescription: {
            type: 'string',
          },
          branchAddress: {
            type: 'string',
          },
          branchPICName: {
            type: 'string',
          },
          orderNo: {
            type: 'string',
          },
          orderStatus: {
            type: 'integer',
          },
          orderType: {
            type: 'integer',
          },
          totalAmount: {
            type: 'integer',
          },
          serviceOrderJobs: {
            type: 'array',
            items: {
              type: 'object',
              required: [
                'jobCode',
                'jobDescription',
              ],
              properties: {
                jobCode: {
                  type: 'string',
                },
                jobDescription: {
                  type: 'string',
                },
              },
            },
          },
          orderStartTime: {
            type: 'string',
          },
          orderEndTime: {
            type: 'string',
          },
          policeRegNo: {
            type: 'string',
          },
          vin: {
            type: 'string',
          },
          vehicleType: {
            type: 'string',
          },
          year: {
            type: 'integer',
          },
          name: {
            type: 'string',
          },
          phone1: {
            type: 'string',
          },
          phone2: {
            type: 'string',
          },
          address: {
            type: 'string',
          },
          serviceAddress: {
            type: 'string',
          },
          notes: {
            type: 'string',
          },
          dscProcessStatus: {
            type: 'integer',
          },
          actualFinishTime: {
            type: 'string',
          },
          isEligibleForRating: {
            type: 'boolean',
          },
          createdTime: {
            type: 'string',
          },
        },
      },
    },
    acknowledge: {
      type: 'integer',
    },
    version: {
      type: 'string',
    },
    build: {
      type: 'string',
    },
  },
};

export const negativeRequest = {
  type: 'object',
  required: [
    'acknowledge',
    'version',
    'build',
  ],
  properties: {
    acknowledge: {
      type: 'integer',
    },
    version: {
      type: 'string',
    },
    build: {
      type: 'string',
    },
  },
};

export const invalidRequest = {
  type: 'object',
  required: [
    'message',
    'modelState',
  ],
  properties: {
    message: {
      type: 'string',
    },
    modelState: {
      type: 'object',
      required: [
        'request.OrderBy',
      ],
      properties: {
        'request.OrderBy': {
          type: 'array',
          items: {
            type: 'string',
          },
        },
      },
    },
  },
};
