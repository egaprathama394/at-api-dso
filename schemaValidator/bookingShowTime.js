export default {
  type: 'object',
  required: [
    'slot',
    'acknowledge',
    'version',
    'build',
  ],
  properties: {
    slot: {
      type: 'array',
      items: {
        type: 'object',
        required: [
          'timeStart',
          'timeEnd',
        ],
        properties: {
          timeStart: {
            type: 'string',
          },
          timeEnd: {
            type: 'string',
          },
        },
      },
    },
    acknowledge: {
      type: 'integer',
    },
    version: {
      type: 'string',
    },
    build: {
      type: 'string',
    },
  },
};
