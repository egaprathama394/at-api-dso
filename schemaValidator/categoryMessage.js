export default {
  type: 'object',
  required: [
    'messageCategories',
    'acknowledge',
    'version',
    'build',
  ],
  properties: {
    messageCategories: {
      type: 'array',
      items: {
        type: 'object',
        required: [
          'companyCode',
          'slug',
          'name',
          'isActive',
          'id',
          'createdBy',
          'createdTime',
          'rowVersion',
        ],
        properties: {
          companyCode: {
            type: 'string',
          },
          slug: {
            type: 'string',
          },
          name: {
            type: 'string',
          },
          isActive: {
            type: 'boolean',
          },
          id: {
            type: 'integer',
          },
          createdBy: {
            type: 'string',
          },
          createdTime: {
            type: 'string',
          },
          rowVersion: {
            type: 'string',
          },
        },
      },
    },
    acknowledge: {
      type: 'integer',
    },
    version: {
      type: 'string',
    },
    build: {
      type: 'string',
    },
  },
};

export const negativeRequest = {
  type: 'object',
  required: [
    'error',
    'error_description',
  ],
  properties: {
    error: {
      type: 'string',
    },
    error_description: {
      type: 'string',
    },
  },
};
