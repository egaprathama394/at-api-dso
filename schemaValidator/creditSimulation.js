export default {
  type: 'object',
  required: [
    'outData',
    'acknowledge',
    'message',
    'version',
    'build',
  ],
  properties: {
    outData: {
      type: 'array',
      items: {
        type: 'object',
        required: [
          'tenor',
          'totalPayFirst',
          'amountInstallment',
          'lastInstallment',
          'guid',
        ],
        properties: {
          tenor: {
            type: 'string',
          },
          totalPayFirst: {
            type: 'string',
          },
          amountInstallment: {
            type: 'string',
          },
          lastInstallment: {
            type: 'string',
          },
          guid: {
            type: 'string',
          },
        },
      },
    },
  },
  acknowledge: {
    type: 'number',
  },
  message: {
    type: 'string',
  },
  version: {
    type: 'string',
  },
  build: {
    type: 'string',
  },
};
