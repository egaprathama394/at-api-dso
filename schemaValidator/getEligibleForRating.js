export default {
  type: 'object',
  required: [
    'unratedOrdersCount',
  ],
  properties: {
    unratedOrdersCount: {
      type: 'integer',
    },
  },
};

export const negativeRequest = {
  type: 'object',
  properties: {
    message: {
      type: 'string',
    },
    modelState: {
      type: 'object',
      properties: {
        'userId.String': {
          type: 'array',
          items: {
            type: 'string',
          },
        },
      },
    },
  },
};
