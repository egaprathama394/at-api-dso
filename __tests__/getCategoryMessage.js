import getCategoryMessage from '../endpoints/categoryMessage';
import * as categoryMessageSchema from '../schemaValidator/categoryMessage';

const fs = require('fs');
const { matchers } = require('jest-json-schema');

expect.extend(matchers);

const ambilToken = fs.readFileSync('./token.json');
const objToken = JSON.parse(ambilToken);

const token = objToken.edit;

describe('Get Category Message Contact Us', () => {
  test('Can get all category message at contact us successfully using authorized token - positive test cases', async () => {
    const response = await getCategoryMessage(token);
    expect(response.status).toEqual(200);
    expect(response.body).toMatchSchema(categoryMessageSchema.default);
    expect(response.body).toMatchSnapshot();
  });
  test('Can not get all category message at contact us using unauthorized token - negative test cases', async () => {
    const response = await getCategoryMessage('xxx');
    expect(response.status).toEqual(401);
    expect(response.body).toMatchSchema(categoryMessageSchema.negativeRequest);
    expect(response.body).toMatchSnapshot();
  });
  test('Can not get all category message at contact us using empty token - negative test cases', async () => {
    const response = await getCategoryMessage('');
    expect(response.status).toEqual(401);
    expect(response.body).toMatchSchema(categoryMessageSchema.negativeRequest);
    expect(response.body).toMatchSnapshot();
  });
});
