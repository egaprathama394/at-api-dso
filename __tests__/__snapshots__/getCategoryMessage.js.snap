// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Get Category Message Contact Us Can get all category message at contact us successfully using authorized token - positive test cases 1`] = `
Object {
  "acknowledge": 1,
  "build": "0",
  "messageCategories": Array [
    Object {
      "companyCode": "0003",
      "createdBy": "SYSTEM",
      "createdTime": "2018-07-16 03:42:30",
      "id": 33,
      "isActive": true,
      "modifiedBy": "cmsdsoadmin",
      "modifiedTime": "2019-06-11 13:06:05",
      "name": "Keluhan Ditambah",
      "rowVersion": "AAAAAAA21yA=",
      "slug": "Keluhan Ditambah",
    },
    Object {
      "companyCode": "0003",
      "createdBy": "SYSTEM",
      "createdTime": "2018-07-16 03:42:30",
      "id": 34,
      "isActive": true,
      "modifiedBy": "cmsdsoadmin",
      "modifiedTime": "2019-05-03 11:08:52",
      "name": "Layanan Purna Jual",
      "rowVersion": "AAAAAAArUnQ=",
      "slug": "Layanan Purna Jual",
    },
    Object {
      "companyCode": "0003",
      "createdBy": "SYSTEM",
      "createdTime": "2018-07-16 03:42:30",
      "id": 35,
      "isActive": true,
      "modifiedBy": "cmsdsoadmin",
      "modifiedTime": "2019-05-03 11:09:43",
      "name": "Pembelian Kendaraan Baru",
      "rowVersion": "AAAAAAArUno=",
      "slug": "Pembelian Kendaraan Baru",
    },
    Object {
      "companyCode": "0003",
      "createdBy": "SYSTEM",
      "createdTime": "2018-07-16 03:42:30",
      "id": 36,
      "isActive": true,
      "modifiedBy": "cmsdsoadmin",
      "modifiedTime": "2019-05-03 19:37:17",
      "name": "Lain-lain1",
      "rowVersion": "AAAAAAArU4k=",
      "slug": "Lain-lain1",
    },
    Object {
      "companyCode": "0003",
      "createdBy": "fabianus.hendyevan@ai.astra.co.id",
      "createdTime": "2018-10-11 14:15:15",
      "id": 42,
      "isActive": true,
      "modifiedBy": "cmsdsoadmin",
      "modifiedTime": "2019-05-03 19:37:33",
      "name": "Membeli Mobil",
      "rowVersion": "AAAAAAArU40=",
      "slug": "Membeli Mobil",
    },
    Object {
      "companyCode": "0003",
      "createdBy": "cmsdsoadmin",
      "createdTime": "2018-10-31 10:34:36",
      "id": 98,
      "isActive": true,
      "modifiedBy": "cmsdsoadmin",
      "modifiedTime": "2018-10-31 11:25:53",
      "name": "Konsultasi",
      "rowVersion": "AAAAAAAPsNg=",
      "slug": "Konsultasi",
    },
    Object {
      "companyCode": "0003",
      "createdBy": "cmsdsoadmin",
      "createdTime": "2018-10-31 14:18:57",
      "id": 99,
      "isActive": true,
      "modifiedBy": "cmsdsoadmin",
      "modifiedTime": "2018-10-31 14:23:35",
      "name": "Informasi",
      "rowVersion": "AAAAAAAPv8s=",
      "slug": "Informasi",
    },
    Object {
      "companyCode": "0003",
      "createdBy": "cmsdsoadmin",
      "createdTime": "2019-01-03 11:38:03",
      "id": 108,
      "isActive": true,
      "modifiedBy": "cmsdsoadmin",
      "modifiedTime": "2019-03-27 15:41:33",
      "name": "Layanan Bengkel",
      "rowVersion": "AAAAAAApq1k=",
      "slug": "Layanan Bengkel",
    },
    Object {
      "companyCode": "0003",
      "createdBy": "cmsdsoadmin",
      "createdTime": "2019-02-20 13:49:47",
      "id": 134,
      "isActive": true,
      "modifiedBy": "cmsdsoadmin",
      "modifiedTime": "2019-03-10 12:29:10",
      "name": "cobarett",
      "rowVersion": "AAAAAAAoypk=",
      "slug": "cobarett",
    },
    Object {
      "companyCode": "0003",
      "createdBy": "cmsdsoadmin",
      "createdTime": "2019-02-20 15:20:01",
      "id": 135,
      "isActive": true,
      "modifiedBy": "cmsdsoadmin",
      "modifiedTime": "2019-02-20 15:23:19",
      "name": "Buy Car Xenia",
      "rowVersion": "AAAAAAAnGrI=",
      "slug": "Buy Car Xenia",
    },
    Object {
      "companyCode": "0003",
      "createdBy": "cmsdsoadmin",
      "createdTime": "2019-03-08 15:07:18",
      "id": 143,
      "isActive": true,
      "modifiedBy": "cmsdsoadmin",
      "modifiedTime": "2019-03-08 15:08:39",
      "name": "booking service",
      "rowVersion": "AAAAAAAoyj4=",
      "slug": "booking service",
    },
    Object {
      "companyCode": "0003",
      "createdBy": "cmsdsoadmin",
      "createdTime": "2019-03-10 17:16:25",
      "id": 145,
      "isActive": true,
      "modifiedBy": "cmsdsoadmin",
      "modifiedTime": "2019-03-10 18:02:03",
      "name": "Ini Category Testing",
      "rowVersion": "AAAAAAAoyy8=",
      "slug": "",
    },
    Object {
      "companyCode": "0003",
      "createdBy": "cmsdsoadmin",
      "createdTime": "2019-03-11 21:46:06",
      "id": 146,
      "isActive": true,
      "name": "Contoh Kategori",
      "rowVersion": "AAAAAAAoziI=",
      "slug": "",
    },
    Object {
      "companyCode": "0003",
      "createdBy": "cmsdsoadmin",
      "createdTime": "2019-03-11 21:52:24",
      "id": 148,
      "isActive": true,
      "name": "Contoh Kategori 3",
      "rowVersion": "AAAAAAAoziQ=",
      "slug": "",
    },
    Object {
      "companyCode": "0003",
      "createdBy": "cmsdsoadmin",
      "createdTime": "2019-03-11 21:56:10",
      "id": 149,
      "isActive": true,
      "name": "Contoh Kategori 5",
      "rowVersion": "AAAAAAAoziU=",
      "slug": "",
    },
    Object {
      "companyCode": "0003",
      "createdBy": "cmsdsoadmin",
      "createdTime": "2019-03-11 21:56:55",
      "id": 150,
      "isActive": true,
      "name": "Contoh Kategori 6",
      "rowVersion": "AAAAAAAoziY=",
      "slug": "",
    },
    Object {
      "companyCode": "0003",
      "createdBy": "cmsdsoadmin",
      "createdTime": "2019-03-11 21:58:54",
      "id": 151,
      "isActive": true,
      "name": "Contoh Kategori 7",
      "rowVersion": "AAAAAAAozic=",
      "slug": "",
    },
    Object {
      "companyCode": "0003",
      "createdBy": "cmsdsoadmin",
      "createdTime": "2019-03-11 21:59:45",
      "id": 152,
      "isActive": true,
      "name": "Contoh Kategori 8",
      "rowVersion": "AAAAAAAozig=",
      "slug": "",
    },
    Object {
      "companyCode": "0003",
      "createdBy": "cmsdsoadmin",
      "createdTime": "2019-03-11 22:01:59",
      "id": 153,
      "isActive": true,
      "name": "Fix Kategori",
      "rowVersion": "AAAAAAAozik=",
      "slug": "",
    },
    Object {
      "companyCode": "0003",
      "createdBy": "cmsdsoadmin",
      "createdTime": "2019-03-11 22:03:03",
      "id": 154,
      "isActive": true,
      "name": "Kredit Mobile Service",
      "rowVersion": "AAAAAAAozio=",
      "slug": "",
    },
    Object {
      "companyCode": "0003",
      "createdBy": "cmsdsoadmin",
      "createdTime": "2019-03-11 22:04:20",
      "id": 155,
      "isActive": true,
      "name": "Daihatsu Promo",
      "rowVersion": "AAAAAAAozis=",
      "slug": "",
    },
    Object {
      "companyCode": "0003",
      "createdBy": "cmsdsoadmin",
      "createdTime": "2019-03-11 22:04:44",
      "id": 156,
      "isActive": true,
      "name": "Promo Daihatsu",
      "rowVersion": "AAAAAAAoziw=",
      "slug": "",
    },
    Object {
      "companyCode": "0003",
      "createdBy": "cmsdsoadmin",
      "createdTime": "2019-03-11 22:09:30",
      "id": 157,
      "isActive": true,
      "modifiedBy": "cmsdsoadmin",
      "modifiedTime": "2019-03-11 23:19:06",
      "name": "Discount",
      "rowVersion": "AAAAAAAozlY=",
      "slug": "",
    },
    Object {
      "companyCode": "0003",
      "createdBy": "cmsdsoadmin",
      "createdTime": "2019-03-11 22:10:44",
      "id": 158,
      "isActive": true,
      "modifiedBy": "cmsdsoadmin",
      "modifiedTime": "2019-03-11 23:18:40",
      "name": "Sport Car",
      "rowVersion": "AAAAAAAozlQ=",
      "slug": "",
    },
    Object {
      "companyCode": "0003",
      "createdBy": "cmsdsoadmin",
      "createdTime": "2019-03-13 10:31:57",
      "id": 160,
      "isActive": true,
      "name": "TEST B",
      "rowVersion": "AAAAAAAo0xw=",
      "slug": "",
    },
    Object {
      "companyCode": "0003",
      "createdBy": "cmsdsoadmin",
      "createdTime": "2019-03-13 16:43:14",
      "id": 162,
      "isActive": true,
      "name": "INI SLUG LOH",
      "rowVersion": "AAAAAAAo1+M=",
      "slug": "",
    },
    Object {
      "companyCode": "0003",
      "createdBy": "cmsdsoadmin",
      "createdTime": "2019-03-13 16:44:17",
      "id": 163,
      "isActive": true,
      "name": "INI SLUG LAGI DEH",
      "rowVersion": "AAAAAAAo1+Y=",
      "slug": "",
    },
    Object {
      "companyCode": "0003",
      "createdBy": "cmsdsoadmin",
      "createdTime": "2019-03-13 16:45:30",
      "id": 164,
      "isActive": true,
      "modifiedBy": "cmsdsoadmin",
      "modifiedTime": "2019-03-13 16:46:01",
      "name": "ANISTIA MAHRANI slug 2",
      "rowVersion": "AAAAAAAo1+o=",
      "slug": "ANISTIA MAHRANI slug 2",
    },
    Object {
      "companyCode": "0003",
      "createdBy": "cmsdsoadmin",
      "createdTime": "2019-03-13 16:47:00",
      "id": 165,
      "isActive": true,
      "modifiedBy": "cmsdsoadmin",
      "modifiedTime": "2019-03-14 15:05:06",
      "name": "INI SLUG TESTING",
      "rowVersion": "AAAAAAAo3Gs=",
      "slug": "",
    },
    Object {
      "companyCode": "0003",
      "createdBy": "cmsdsoadmin",
      "createdTime": "2019-03-13 17:02:47",
      "id": 166,
      "isActive": true,
      "modifiedBy": "cmsdsoadmin",
      "modifiedTime": "2019-03-13 17:20:12",
      "name": "Cobadites",
      "rowVersion": "AAAAAAAo2E4=",
      "slug": "Cobadites",
    },
    Object {
      "companyCode": "0003",
      "createdBy": "cmsdsoadmin",
      "createdTime": "2019-03-13 17:44:54",
      "id": 167,
      "isActive": true,
      "name": "Slug fix v1",
      "rowVersion": "AAAAAAAo2FA=",
      "slug": "",
    },
    Object {
      "companyCode": "0003",
      "createdBy": "cmsdsoadmin",
      "createdTime": "2019-03-13 17:47:55",
      "id": 168,
      "isActive": true,
      "name": "Slug fix v2",
      "rowVersion": "AAAAAAAo2FM=",
      "slug": "Slug fix v2",
    },
    Object {
      "companyCode": "0003",
      "createdBy": "cmsdsoadmin",
      "createdTime": "2019-03-13 19:05:08",
      "id": 169,
      "isActive": true,
      "name": "TESTING CATEGODY",
      "rowVersion": "AAAAAAAo2Gg=",
      "slug": "",
    },
    Object {
      "companyCode": "0003",
      "createdBy": "cmsdsoadmin",
      "createdTime": "2019-03-13 19:07:09",
      "id": 170,
      "isActive": true,
      "name": "Ini Category Testing 3",
      "rowVersion": "AAAAAAAo2Gw=",
      "slug": "",
    },
    Object {
      "companyCode": "0003",
      "createdBy": "cmsdsoadmin",
      "createdTime": "2019-03-13 19:08:15",
      "id": 171,
      "isActive": true,
      "modifiedBy": "cmsdsoadmin",
      "modifiedTime": "2019-03-14 15:10:48",
      "name": "ANISTIA MAHRANI iu",
      "rowVersion": "AAAAAAAo3HQ=",
      "slug": "",
    },
    Object {
      "companyCode": "0003",
      "createdBy": "cmsdsoadmin",
      "createdTime": "2019-03-14 10:38:32",
      "id": 173,
      "isActive": true,
      "name": "CategoryA",
      "rowVersion": "AAAAAAAo2iM=",
      "slug": "CategoryA",
    },
    Object {
      "companyCode": "0003",
      "createdBy": "cmsdsoadmin",
      "createdTime": "2019-03-14 10:54:02",
      "id": 174,
      "isActive": true,
      "name": "Category B",
      "rowVersion": "AAAAAAAo2kk=",
      "slug": "Category B",
    },
    Object {
      "companyCode": "0003",
      "createdBy": "cmsdsoadmin",
      "createdTime": "2019-03-14 10:54:26",
      "id": 175,
      "isActive": true,
      "name": "Category C",
      "rowVersion": "AAAAAAAo2ko=",
      "slug": "Category C",
    },
    Object {
      "companyCode": "0003",
      "createdBy": "cmsdsoadmin",
      "createdTime": "2019-03-14 10:54:32",
      "id": 176,
      "isActive": true,
      "modifiedBy": "cmsdsoadmin",
      "modifiedTime": "2019-03-14 14:34:25",
      "name": "Category D",
      "rowVersion": "AAAAAAAo29M=",
      "slug": "Category D",
    },
    Object {
      "companyCode": "0003",
      "createdBy": "cmsdsoadmin",
      "createdTime": "2019-03-14 15:10:37",
      "id": 177,
      "isActive": true,
      "name": "Category Z",
      "rowVersion": "AAAAAAAo3HE=",
      "slug": "Category Z",
    },
    Object {
      "companyCode": "0003",
      "createdBy": "cmsdsoadmin",
      "createdTime": "2019-03-14 15:11:22",
      "id": 178,
      "isActive": true,
      "name": "Category Y",
      "rowVersion": "AAAAAAAo3HU=",
      "slug": "Category Y",
    },
    Object {
      "companyCode": "0003",
      "createdBy": "cmsdsoadmin",
      "createdTime": "2019-03-14 15:12:47",
      "id": 179,
      "isActive": true,
      "name": "IYA NAMBAH",
      "rowVersion": "AAAAAAAo3Hg=",
      "slug": "IYA NAMBAH",
    },
    Object {
      "companyCode": "0003",
      "createdBy": "cmsdsoadmin",
      "createdTime": "2019-03-14 15:55:50",
      "id": 181,
      "isActive": true,
      "name": "Category X",
      "rowVersion": "AAAAAAAo3K8=",
      "slug": "Category X",
    },
    Object {
      "companyCode": "0003",
      "createdBy": "cmsdsoadmin",
      "createdTime": "2019-03-14 16:29:24",
      "id": 182,
      "isActive": true,
      "modifiedBy": "cmsdsoadmin",
      "modifiedTime": "2019-03-14 16:50:28",
      "name": "HelloWorld",
      "rowVersion": "AAAAAAAo3TA=",
      "slug": "HelloWorld",
    },
    Object {
      "companyCode": "0003",
      "createdBy": "cmsdsoadmin",
      "createdTime": "2019-03-14 16:49:20",
      "id": 183,
      "isActive": true,
      "name": "football",
      "rowVersion": "AAAAAAAo3R8=",
      "slug": "football",
    },
    Object {
      "companyCode": "0003",
      "createdBy": "cmsdsoadmin",
      "createdTime": "2019-03-14 17:08:22",
      "id": 184,
      "isActive": true,
      "name": "Category V",
      "rowVersion": "AAAAAAAo3cI=",
      "slug": "Category V",
    },
    Object {
      "companyCode": "0003",
      "createdBy": "cmsdsoadmin",
      "createdTime": "2019-03-15 16:26:31",
      "id": 185,
      "isActive": true,
      "modifiedBy": "cmsdsoadmin",
      "modifiedTime": "2019-03-21 14:39:52",
      "name": "OKHAY",
      "rowVersion": "AAAAAAApKWw=",
      "slug": "OKHAY",
    },
    Object {
      "companyCode": "0003",
      "createdBy": "cmsdsoadmin",
      "createdTime": "2019-03-21 14:40:06",
      "id": 186,
      "isActive": true,
      "name": "YAA",
      "rowVersion": "AAAAAAApKW0=",
      "slug": "YAA",
    },
    Object {
      "companyCode": "0003",
      "createdBy": "cmsdsoadmin",
      "createdTime": "2019-03-21 16:29:45",
      "id": 187,
      "isActive": true,
      "name": "Anime",
      "rowVersion": "AAAAAAApKdQ=",
      "slug": "Anime",
    },
    Object {
      "companyCode": "0003",
      "createdBy": "cmsdsoadmin",
      "createdTime": "2019-03-25 10:49:59",
      "id": 188,
      "isActive": true,
      "name": "Test2",
      "rowVersion": "AAAAAAApSuU=",
      "slug": "Test2",
    },
    Object {
      "companyCode": "0003",
      "createdBy": "cmsdsoadmin",
      "createdTime": "2019-03-25 10:54:15",
      "id": 189,
      "isActive": true,
      "name": "fiqih",
      "rowVersion": "AAAAAAApSu8=",
      "slug": "fiqih",
    },
    Object {
      "companyCode": "0003",
      "createdBy": "cmsdsoadmin",
      "createdTime": "2019-03-25 11:51:40",
      "id": 190,
      "isActive": true,
      "name": "Sample 5",
      "rowVersion": "AAAAAAApSwQ=",
      "slug": "Sample 5",
    },
    Object {
      "companyCode": "0003",
      "createdBy": "cmsdsoadmin",
      "createdTime": "2019-03-25 13:38:32",
      "id": 191,
      "isActive": true,
      "name": "contoh",
      "rowVersion": "AAAAAAApS0I=",
      "slug": "contoh",
    },
    Object {
      "companyCode": "0003",
      "createdBy": "cmsdsoadmin",
      "createdTime": "2019-03-25 13:52:32",
      "id": 192,
      "isActive": true,
      "name": "ya",
      "rowVersion": "AAAAAAApbCs=",
      "slug": "ya",
    },
    Object {
      "companyCode": "0003",
      "createdBy": "cmsdsoadmin",
      "createdTime": "2019-03-25 14:01:18",
      "id": 193,
      "isActive": true,
      "name": "Tidak",
      "rowVersion": "AAAAAAApkGk=",
      "slug": "Tidak",
    },
    Object {
      "companyCode": "0003",
      "createdBy": "SYSTEM",
      "createdTime": "2019-03-26 18:29:26",
      "id": 194,
      "isActive": true,
      "name": "Keluhan1",
      "rowVersion": "AAAAAAApqBk=",
      "slug": "Keluhan1",
    },
    Object {
      "companyCode": "0003",
      "createdBy": "SYSTEM",
      "createdTime": "2019-03-26 18:34:30",
      "id": 195,
      "isActive": true,
      "name": "Keluhan12",
      "rowVersion": "AAAAAAApqBs=",
      "slug": "Keluhan12",
    },
    Object {
      "companyCode": "0003",
      "createdBy": "SYSTEM",
      "createdTime": "2019-03-26 18:49:45",
      "id": 196,
      "isActive": true,
      "name": "Keluhan123",
      "rowVersion": "AAAAAAApqBw=",
      "slug": "Keluhan123",
    },
    Object {
      "companyCode": "0003",
      "createdBy": "SYSTEM",
      "createdTime": "2019-03-26 18:51:30",
      "id": 197,
      "isActive": true,
      "name": "Keluhan12344",
      "rowVersion": "AAAAAAApqB0=",
      "slug": "Keluhan12344",
    },
    Object {
      "companyCode": "0003",
      "createdBy": "SYSTEM",
      "createdTime": "2019-03-26 18:55:25",
      "id": 198,
      "isActive": true,
      "name": "aaKeluhan1234334",
      "rowVersion": "AAAAAAApqB4=",
      "slug": "aaKeluhan1234334",
    },
    Object {
      "companyCode": "0003",
      "createdBy": "SYSTEM",
      "createdTime": "2019-03-26 18:56:22",
      "id": 199,
      "isActive": true,
      "name": "aaKeluhan1234334",
      "rowVersion": "AAAAAAApqB8=",
      "slug": "aaKeluhan1234334",
    },
    Object {
      "companyCode": "0003",
      "createdBy": "cmsdsoadmin",
      "createdTime": "2019-04-23 10:25:59",
      "id": 210,
      "isActive": true,
      "name": "testes",
      "rowVersion": "AAAAAAAp85Q=",
      "slug": "testes",
    },
    Object {
      "companyCode": "0003",
      "createdBy": "cmsdsoadmin",
      "createdTime": "2019-04-23 10:26:13",
      "id": 211,
      "isActive": true,
      "name": "Keluhan Pelanggan",
      "rowVersion": "AAAAAAAp85Y=",
      "slug": "Keluhan Pelanggan",
    },
    Object {
      "companyCode": "0003",
      "createdBy": "cmsdsoadmin",
      "createdTime": "2019-04-24 19:39:45",
      "id": 214,
      "isActive": true,
      "name": "tambah add maintain april2019",
      "rowVersion": "AAAAAAAp9N4=",
      "slug": "tambah add maintain april2019",
    },
    Object {
      "companyCode": "0003",
      "createdBy": "cmsdsoadmin",
      "createdTime": "2019-04-24 19:49:58",
      "id": 215,
      "isActive": true,
      "name": "april2019",
      "rowVersion": "AAAAAAAp9OM=",
      "slug": "april2019",
    },
    Object {
      "companyCode": "0003",
      "createdBy": "cmsdsoadmin",
      "createdTime": "2019-04-24 20:01:27",
      "id": 216,
      "isActive": true,
      "name": "yaa",
      "rowVersion": "AAAAAAAp9OY=",
      "slug": "yaa",
    },
    Object {
      "companyCode": "0003",
      "createdBy": "cmsdsoadmin",
      "createdTime": "2019-05-03 17:11:28",
      "id": 218,
      "isActive": true,
      "modifiedBy": "cmsdsoadmin",
      "modifiedTime": "2019-05-03 19:39:45",
      "name": "ANISTIA MAHRANI LAGI",
      "rowVersion": "AAAAAAArU5Y=",
      "slug": "ANISTIA MAHRANI LAGI",
    },
    Object {
      "companyCode": "0003",
      "createdBy": "cmsdsoadmin",
      "createdTime": "2019-05-04 16:09:10",
      "id": 219,
      "isActive": true,
      "name": "Percobaan",
      "rowVersion": "AAAAAAArVGY=",
      "slug": "Percobaan",
    },
  ],
  "version": "1.1",
}
`;

exports[`Get Category Message Contact Us Can not get all category message at contact us using empty token - negative test cases 1`] = `
Object {
  "error": "invalid_token",
  "error_description": "The access token is missing.",
}
`;

exports[`Get Category Message Contact Us Can not get all category message at contact us using unauthorized token - negative test cases 1`] = `
Object {
  "error": "invalid_token",
  "error_description": "The access token is invalid or expired.",
}
`;
